sql/core/src/main/scala/org/apache/spark/sql/execution/GeneratedAggregate.scala

```diff
@@ -241,7 +241,7 @@ case class GeneratedAggregate(
     child.execute().mapPartitions { iter =>
       // Builds a new custom class for holding the results of aggregation for a group.
       val initialValues = computeFunctions.flatMap(_.initialValues)
-      val newAggregationBuffer = newProjection(initialValues, child.output)
+      val newAggregationBuffer = newProjection(initialValues, child.output, mutableRow = true)
```

Comment:
We wrote `newAggregationBuffer(EmptyRow).asInstanceOf[MutableRow]` below, I think that is the root case. cc @davies